src/dialogxml/widgets/field.hpp

```diff
@@ -90,6 +90,10 @@ class cTextField : public cControl {
 	rectangle text_rect;
 	std::vector<snippet_t> snippets;
 	int ip_row, ip_col;
+	/// Setting maxChars AFTER the player has a chance to type in the field would be a bad idea.
+	/// So would calling setText() with a string longer than maxChars.
+	/// Really, it should probably only be set in xml, as the attribute "max-chars".
+	int maxChars = -1;
```

Comment:
I wouldn't call it a priority to have a runtime setter for `maxChars`, but I don't think it's something that's difficult to do – it could just truncate the current text to the new maximum length.